filter/filter.go

```diff
@@ -0,0 +1,45 @@
+package filter
+
+import (
+	"net/http"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/http-proxy/utils"
+)
+
+var (
+	log = golog.LoggerFor("http-proxy.filter")
+)
+
+// Filter is a special http.Handler that returns true or false depending on
+// whether subsequent handlers should continue.
+type Filter interface {
+	// ServeHTTP is like the function on http.Handler but also returns true or
+	// false depending on whether subsequent handlers should continue.
+	ServeHTTP(w http.ResponseWriter, req *http.Request) (ok bool, err error)
+}
+
+// filterChain is a chain of filters that implements the http.Handler
+// interface.
+type filterChain struct {
+	filters []Filter
+}
+
+// NewChain constructs a new chain of filters that executes the filters in order
+// until it encounters a filter that returns false.
+func NewChain(filters ...Filter) http.Handler {
+	return &filterChain{filters}
+}
+
+func (chain *filterChain) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	for _, filter := range chain.filters {
+		ok, err := filter.ServeHTTP(w, req)
+		if err != nil {
+			log.Debugf("Encountered error in filter: %v", err)
+			utils.DefaultHandler.ServeHTTP(w, req, err)
```

Comment:
Instead of having each filter hang on to utils.DefaultHandler, the filters just return their errors an they get handled in this single place, consistently.